### PR TITLE
HYDRA-824 - add unit test for maya lights

### PR DESCRIPTION
**Pull request by lilike-adsk**
Modify the testDirectionalLights test to cover more cases for MAYA lights:
1.different lighting modes
2.shadow on/off

### CODE REVIEW

**Comment by lilike-adsk** [comment]
FYI @debloip-adsk @roopavr-adsk @ppt-adsk @vlasovi

File: test/lib/mayaUsd/render/mayaToHydra/MayaLightsTest/flatLight.png
Comment (lanierd-adsk) [inline test/lib/mayaUsd/render/mayaToHydra/MayaLightsTest/flatLight.png:1]: This image is exactly the same as with Shadows on. But I guess it's broken ? Is that the expected result ?

Comment (lanierd-adsk) [reply test/lib/mayaUsd/render/mayaToHydra/MayaLightsTest/flatLight.png:1]: It is the same as nolight.png. I think there is a problem.

File: test/lib/mayaUsd/render/mayaToHydra/MayaLightsTest/noLight_shadowOn.png
Comment (lanierd-adsk) [inline test/lib/mayaUsd/render/mayaToHydra/MayaLightsTest/noLight_shadowOn.png:1]: If you have a scene with no light, I think it's not necessary to render an image without shadows. You could remove that test which is exactly the same image as the nolight.png

File: test/lib/mayaUsd/render/mayaToHydra/MayaLightsTest/pointLight_shadowOn.png
Comment (lanierd-adsk) [inline test/lib/mayaUsd/render/mayaToHydra/MayaLightsTest/pointLight_shadowOn.png:1]: I don't see any difference visually between the pointLight.png file, is there any ?

File: test/lib/mayaUsd/render/mayaToHydra/MayaLightsTest/spotLight_shadowOn.png
Comment (lanierd-adsk) [inline test/lib/mayaUsd/render/mayaToHydra/MayaLightsTest/spotLight_shadowOn.png:1]: Same here, no difference between this image and spotLight.png. Is that expected ?

File: test/lib/mayaUsd/render/mayaToHydra/testMayaLights.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+# limitations under the License.
+#
+
+import maya.cmds as cmds
+import fixturesUtils
+import mtohUtils
+import mayaUtils
+import unittest
+
+class TesMayatLights(mtohUtils.MtohTestCase): #Subclassing mtohUtils.MtohTestCase to be able to call self.assertSnapshotClose
```
Comment (lanierd-adsk) [inline test/lib/mayaUsd/render/mayaToHydra/testMayaLights.py]: The "t" from test has been moved after Maya in the class name. :-)  Can you please fix it ? Thanks.

File: test/lib/mayaUsd/render/mayaToHydra/testMayaLights.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        cmds.file(new=True, force=True)
+        cmds.refresh()
+
+        # Load a maya scene with a maya native sphere, usd sphere and some lights, with HdStorm already being the viewport renderer.
+        # The sphere is not at the origin on purpose
+        testFile = mayaUtils.openTestScene(
+                "testMayaLights",
+                "testMayaLights.ma")
+        cmds.refresh()
+        #Do a view fit --Test removed as on Linux the fit doesn't produce the same result as on Windows and OSX.
```
Comment (lanierd-adsk) [inline test/lib/mayaUsd/render/mayaToHydra/testMayaLights.py]: Can you please remove that commented code which I did in a previous PR, it's not required to keep it. Thanks.

**Review by lanierd-adsk — CHANGES_REQUESTED** [review]

File: test/lib/mayaUsd/render/mayaToHydra/MayaLightsTest/flatLight.png
Comment (lilike-adsk) [reply test/lib/mayaUsd/render/mayaToHydra/MayaLightsTest/flatLight.png:1]: flatLight won't create shadows, that's expected.
flatLight results (constant shading) should be different vs. noLight, this seems to be broken or never works.

We can leave the images here, and rebaseline when the issue is fixed.

File: test/lib/mayaUsd/render/mayaToHydra/MayaLightsTest/noLight_shadowOn.png
Comment (lilike-adsk) [reply test/lib/mayaUsd/render/mayaToHydra/MayaLightsTest/noLight_shadowOn.png:1]: I'm fine to remove that case.

File: test/lib/mayaUsd/render/mayaToHydra/MayaLightsTest/pointLight_shadowOn.png
Comment (lilike-adsk) [reply test/lib/mayaUsd/render/mayaToHydra/MayaLightsTest/pointLight_shadowOn.png:1]: This seems to be broken or never works.

File: test/lib/mayaUsd/render/mayaToHydra/MayaLightsTest/flatLight.png
Comment (lanierd-adsk) [reply test/lib/mayaUsd/render/mayaToHydra/MayaLightsTest/flatLight.png:1]: I would prefer removing the wrong images of what is not working and comment the code producing the images in this test and add a JIRA so we fix/support it later. Thanks !

File: test/lib/mayaUsd/render/mayaToHydra/MayaLightsTest/pointLight_shadowOn.png
Comment (lanierd-adsk) [reply test/lib/mayaUsd/render/mayaToHydra/MayaLightsTest/pointLight_shadowOn.png:1]: I would prefer removing the wrong images of what is not working and comment the code producing the images in this test and add a JIRA so we fix/support it later. Thanks !

**Review by lanierd-adsk — APPROVED** [review]

**Pull request merged**